manywheel/build_rocm.sh

```diff
@@ -223,56 +213,86 @@ HIPBLASLT_LIB_FILES=($ARCH_SPECIFIC_FILES $OTHER_FILES)
 # ROCm library files
 # Overwrite ROCM_SO_FILES to contain only libmagma.so if BUILD_LIGHTWEIGHT is enabled
 if [[ "$BUILD_LIGHTWEIGHT" == "1" ]]; then
-    ROCM_SO_FILES=(
+    LIGHTWEIGHT_ROCM_SO_FILES=(
         "libmagma.so"
     )
 fi
-ROCM_SO_PATHS=()
-for lib in "${ROCM_SO_FILES[@]}"
-do
-    file_path=($(find $ROCM_HOME/lib/ -name "$lib")) # First search in lib
-    if [[ -z $file_path ]]; then
-        if [ -d "$ROCM_HOME/lib64/" ]; then
-            file_path=($(find $ROCM_HOME/lib64/ -name "$lib")) # Then search in lib64
+
+do_lightweight_build() {
+    echo "=== Building LIGHTWEIGHT variant ==="
+
+    # First, gather actual file paths for the minimal set
+    ROCM_SO_PATHS_LIGHTWEIGHT=()
+    for lib in "${LIGHTWEIGHT_ROCM_SO_FILES[@]}"; do
+        file_path=($(find $ROCM_HOME/lib/ -name "$lib"))
+        if [[ -z $file_path && -d "$ROCM_HOME/lib64" ]]; then
+             file_path=($(find $ROCM_HOME/lib64/ -name "$lib")) # Then search in lib64
         fi
+        if [[ -z $file_path ]]; then
+            file_path=($(find $ROCM_HOME/ -name "$lib")) # Then search in ROCM_HOME
+        fi
+        if [[ -z $file_path ]]; then
+            echo "Error: Lightweight library $lib is not found." >&2
+            exit 1
+        fi
+	ROCM_SO_PATHS_LIGHTWEIGHT[${#ROCM_SO_PATHS_LIGHTWEIGHT[@]}]="$file_path" # Append lib to array
+    done
+
+    # Set environment so build_common.sh (or build_libtorch.sh) sees it
+    DEPS_LIST=(${ROCM_SO_PATHS_LIGHTWEIGHT[*]})
+    DEPS_SONAME=(${LIGHTWEIGHT_ROCM_SO_FILES[*]})
+    DEPS_AUX_SRCLIST=()
+    DEPS_AUX_DSTLIST=()
+
+    # Because original script doesn't add OS libraries or arch-specific .co
+    # files for "lightweight", we skip them here
+
+    # Finally, source the main build script
+    SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+    if [[ -z "$BUILD_PYTHONLESS" ]]; then
+        BUILD_SCRIPT=build_common.sh
+    else
+        BUILD_SCRIPT=build_libtorch.sh
     fi
-    if [[ -z $file_path ]]; then
-        file_path=($(find $ROCM_HOME/ -name "$lib")) # Then search in ROCM_HOME
-    fi
-    if [[ -z $file_path ]]; then
-        echo "Error: Library file $lib is not found." >&2
-        exit 1
-    fi
-    ROCM_SO_PATHS[${#ROCM_SO_PATHS[@]}]="$file_path" # Append lib to array
-done
+    source "$SCRIPTPATH/${BUILD_SCRIPT}"
 
-DEPS_LIST=(
-    ${ROCM_SO_PATHS[*]}
-)
-if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
-    DEPS_LIST+=(${OS_SO_PATHS[*]})
-fi
-DEPS_SONAME=(
-    ${ROCM_SO_FILES[*]}
-)
-if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
-    DEPS_SONAME+=(${OS_SO_FILES[*]})
```

Comment:
Looks like we totally removed the code that used to append OS_SO_FILES to DEPS_SONAME even for heavyweight case??